lambda-natchez/src/main/scala/feral/lambda/natchez/HasKernel.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feral.lambda.natchez
+
+import feral.lambda.events.ApiGatewayProxyEventV2
+import natchez.Kernel
+
+trait HasKernel[Event] {
+  def extract(event: Event): Kernel
+}
+
+object HasKernel extends HasKernelLowPriority {
+  @inline def apply[E](implicit hk: HasKernel[E]): hk.type = hk
+
+  implicit val apiGateProxyEventV2Kernel: HasKernel[ApiGatewayProxyEventV2] =
+    e => Kernel(e.headers)
+}
```

Comment:
One of the nice things about having all the events together in the same module is it makes it easy to provide these implicits. Although I'm doubtful if any will have a special implementation except this one.